CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -342,9 +342,16 @@ function Patient:die()
 end
 
 function Patient:canPeeOrPuke(current)
-  return ((current.name == "walk" or current.name == "idle" or
-           current.name == "seek_room" or current.name == "queue") and
-      not self.going_home and self.world.map.th:getCellFlags(self.tile_x, self.tile_y).buildable)
+  local good_actions = {walk=true, idle=true, seek_room=true, queue=true}
+  if not good_actions[current] then return false end
```

Comment:
Arguably, 346+347 could be one if statement? Though I'm not against it because of its clarity

Reply:
You're correct that it could be 1 statement, I just tend to prefer multiple shorter and simpler statements.